KHWE-2: setup environment

gamesminer opened the pull request:
setup environment

gamesminer committed:
setup environment

ihor4568 commented on index.html:
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
```
please use webpack html plugin

ihor4568 replied:
why code is not formatted?

ihor4568 commented on package.json:
```diff
@@ -0,0 +1,53 @@
+      "prettier --write",
+      "git add"
+    ],
+    "index.html": [
```
you will have to format every html file - not only index.html

ihor4568 commented on src/index.js:
```diff
@@ -0,0 +1,2 @@
+// CSS files import
```
why do you need this comment?

ihor4568 commented on index.html:
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html lang="eu" class="page">
```
why lang eu ?? lang en

ihor4568 commented on index.html:
```diff
@@ -0,0 +1,20 @@
+
+    <div id="root"></div>
+
+    <script src="dist/main.js"></script>
```
why do you need to add script if you use html webpack plugin?

ihor4568 commented on webpack.config.js:
```diff
@@ -0,0 +1,64 @@
+        },
+        {
+          test: /\.html$/,
+          loader: "html-loader"
```
why do you need html-loader?

ihor4568 commented on webpack.config.js:
```diff
@@ -0,0 +1,64 @@
+        {
+          test: /\.html$/,
+          loader: "html-loader"
+        }
```
please add https://www.npmjs.com/package/ejs-loader  for html extension

ihor4568 commented on index.html:
```diff
@@ -0,0 +1,20 @@
+    <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
+</head>
+
+<body class="body">
```
why do you need this class name ??

ihor4568 commented:
Please add all your team mates as PR reviewers

ihor4568 commented:
The PR name should be like this: KHWE-2: setup environment

ihor4568 commented:
the commits should follow this style: KHWE-2: commit text

gamesminer committed:
Pre-commit fix

gamesminer committed:
Fix lang, <script>

gamesminer committed:
Delete comment

gamesminer committed:
KHWE-2: deleted connection style.css & main.js in index.html; install ejs-loader; deleted html-loader

gamesminer commented:
> Please add all your team mates as PR reviewers

Sorry, but not all my team mates have accepted the request

yevheniiIvanise commented on webpack.config.js:
```diff
@@ -0,0 +1,64 @@
+        }
+      ]
+    },
+    plugins: [
```
Could you setup CleanWebpackPlugin plugin please?

yevheniiIvanise commented on package.json:
```diff
@@ -0,0 +1,53 @@
+    "@material/ripple": "^0.40.0",
+    "autoprefixer": "^9.1.5",
+    "css-loader": "^1.0.0",
+    "extract-text-webpack-plugin": "^4.0.0-beta.0",
```
I would pick MiniCssExtractPlugin instead of this to minify css as webpack docs says. It is still beta for webpack 4

yevheniiIvanise commented on package.json:
```diff
@@ -0,0 +1,53 @@
+      "prettier --write",
+      "git add"
+    ],
+    "*{/*.scss,/*/*.scss}": [
```
you can write it in upper object property within *.{js,json,css,md}

gamesminer committed:
KHWE-2: set ejs-loader, fix index.html with HTML webpack plugin

yevheniiIvanise commented on webpack.config.js:
```diff
@@ -0,0 +1,64 @@
+            {
+              loader: "postcss-loader",
+              options: {
+                plugins: () => [autoprefixer()]
```
why did you create postcss.config.js and described the same here? pick either postscss.config.js or this way

yevheniiIvanise commented on package.json:
```diff
@@ -0,0 +1,53 @@
+    "file-loader": "^2.0.0",
+    "html-webpack-plugin": "^3.2.0",
+    "husky": "^1.0.0-rc.14",
+    "img-loader": "^3.0.0",
```
why do we need it to be installed?) ru gonna use this?

yevheniiIvanise commented on webpack.config.js:
```diff
@@ -0,0 +1,75 @@
+                options: {
+                  sourceMap: true,
+                  plugins: [
+                    autoprefixer({
```
why you create postcss.config.js and mention the same here? pick either postscss.config.js or this way

gamesminer committed:
KHWE-2: Fix package.json&webpack.config; install plugins to webpack

In package.json fixed script "build" because we build prod version and
changed parametrs for lint-staged (combined css, js, scss, json in one
field. Installed and insert to webpack.config CleanWebpackPlugin and
MiniCssExtractPlugin to minify css as webpack docs says. And created
postcss.config.js for adding autoprefixer

ihor4568 commented on index.html:
```diff
@@ -0,0 +1,18 @@
+<html lang="en" class="page">
+
+<head>
+    <meta charset="UTF-8">
```
why does this file has 4 spaces indent? or these are tabs?

ihor4568 commented on index.html:
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html lang="en" class="page">
```
why do you need class name if you do not use it in code?

ihor4568 commented on package.json:
```diff
@@ -0,0 +1,50 @@
+  "keywords": [],
+  "license": "ISC",
+  "devDependencies": {
+    "@material/ripple": "^0.40.0",
```
where do you use this dependency?

ihor4568 replied:
please freeze all deps - I mean - remove `^` before each package version

ihor4568 commented on webpack.config.js:
```diff
@@ -0,0 +1,58 @@
+          use: "file-loader"
+        },
+        {
+          test: /\.ejs$/,
```
please use html extension here

ihor4568 commented on webpack.config.js:
```diff
@@ -0,0 +1,58 @@
+        {
+          test: /\.ejs$/,
+          loader: "ejs-loader",
+          query: {
```
please use default settings - remove query here

ihor4568 commented on webpack.config.js:
```diff
@@ -0,0 +1,58 @@
+    entry: "./src/index.js",
+    output: {
+      path: path.resolve(__dirname, "./dist"),
+      filename: "main.js"
```
you can also use smth like "name].[hash].js" here

yevheniiIvanise commented on webpack.config.js:
```diff
@@ -0,0 +1,58 @@
+      rules: [
+        {
+          test: /\.scss$/,
+          use: ExtractTextPlugin.extract({
```
why do we need both ExtractTextPlugin and MiniCssExtractPlugin?)

ihor4568 commented on package.json:
```diff
@@ -0,0 +1,50 @@
+{
+  "name": "Spotty",
+  "version": "1.0.0",
+  "description": "spotty",
```
let description be "Simple music streaming service"

ihor4568 commented on package.json:
```diff
@@ -0,0 +1,50 @@
+{
```
please also add `private: true` to package.json

yevheniiIvanise commented on package.json:
```diff
@@ -0,0 +1,50 @@
+  "name": "Spotty",
+  "version": "1.0.0",
+  "description": "spotty",
+  "main": "src/js/index.js",
```
i am not sure we need this main entry

ihor4568 replied:
++

gamesminer committed:
KHWE-2: Fix index.html&package.json. Use html extension for ejs

Fix index.html (delete some spaces in paragraphs & class for tag <html>)
Use html extension for ejs-loader & removed query for this.
Deleted ExtractTextPlugin to install MiniCssExtractPlugin.
Fix package.json (change description & deled field "main") & added field
"private: true" and froze all deps

ihor4568 commented on webpack.config.js:
```diff
@@ -0,0 +1,71 @@
+            },
+            {
+              loader: "sass-loader",
+              options: {}
```
why do you need this `options` if it is empty?

gamesminer committed:
Delete empty option sass-loader
